crates/ruff_python_formatter/tests/snapshots/black_compatibility@simple_cases__fmtonoff.py.snap

```diff
@@ -198,192 +198,47 @@ d={'a':1,
 ```diff
 --- Black
 +++ Ruff
-@@ -6,8 +6,8 @@
- 
- from library import some_connection, some_decorator
- # fmt: off
--from third_party import (X,
--                         Y, Z)
-+from third_party import X, Y, Z
-+
- # fmt: on
- f"trigger 3.6 mode"
- # Comment 1
-@@ -17,26 +17,40 @@
- 
- # fmt: off
- def func_no_args():
--  a; b; c
--  if True: raise RuntimeError
--  if False: ...
--  for i in range(10):
--    print(i)
--    continue
--  exec('new-style exec', {}, {})
--  return None
-+    a
-+    b
-+    c
-+    if True:
-+        raise RuntimeError
-+    if False:
-+        ...
-+    for i in range(10):
-+        print(i)
-+        continue
-+    exec("new-style exec", {}, {})
-+    return None
-+
-+
- async def coroutine(arg, exec=False):
-- 'Single-line docstring. Multiline is harder to reformat.'
-- async with some_connection() as conn:
--     await conn.do_what_i_mean('SELECT bobby, tables FROM xkcd', timeout=2)
-- await asyncio.sleep(1)
-+    "Single-line docstring. Multiline is harder to reformat."
-+    async with some_connection() as conn:
-+        await conn.do_what_i_mean("SELECT bobby, tables FROM xkcd", timeout=2)
-+    await asyncio.sleep(1)
-+
-+
- @asyncio.coroutine
--@some_decorator(
--with_args=True,
--many_args=[1,2,3]
--)
--def function_signature_stress_test(number:int,no_annotation=None,text:str='default',* ,debug:bool=False,**kwargs) -> str:
-- return text[number:-1]
-+@some_decorator(with_args=True, many_args=[1, 2, 3])
-+def function_signature_stress_test(
-+    number: int,
-+    no_annotation=None,
-+    text: str = "default",
-+    *,
-+    debug: bool = False,
-+    **kwargs,
-+) -> str:
-+    return text[number:-1]
-+
-+
- # fmt: on
- def spaces(a=1, b=(), c=[], d={}, e=True, f=-1, g=1 if False else 2, h="", i=r""):
-     offset = attr.ib(default=attr.Factory(lambda: _r.uniform(1, 2)))
-@@ -63,15 +77,15 @@
+@@ -63,7 +63,7 @@
  
  something = {
      # fmt: off
 -    key: 'value',
-+    key: "value",
++    key: 'value'
```

Comment:
We incorrectly remove the trailing comma because we don't know that it is part of the value (it has no source information attached). It works as expected if the dictionary contains a `fmt: on` comment.